otx/core/data/adapter/segmentation_dataset_adapter.py

```diff
@@ -92,6 +92,7 @@ def get_otx_dataset(self) -> DatasetEntity:
                                 shapes.append(self._get_polygon_entity(d_polygon, image.width, image.height))
                                 if d_polygon.label not in used_labels:
                                     used_labels.append(d_polygon.label)
+
                     if len(shapes) > 0 or subset == Subset.UNLABELED:
```

Comment:
I don't get why there is no if-statement `len(datumaro_item.annotations) == 0` in seg like det. Does seg include an image file listed in dataset subset but with no annotation well?

Reply:
For segmentation, we need to consider various cases.

If we assume that all datasets have `background` label, then `len(datumaro_item.annotations)==0` shouldn't be applied.

Reply:
In other words, an empty image also is predicted as `background` in the segmentation case. So, all segmentation data items have annotation to represents the background